bcc/module.go

```diff
@@ -114,9 +114,11 @@ func (bpf *Module) Close() {
 	C.bpf_module_destroy(bpf.p)
 	for k, v := range bpf.kprobes {
 		C.perf_reader_free(v)
-		desc := fmt.Sprintf("-:kprobes/%s", k)
-		descCS := C.CString(desc)
-		C.bpf_detach_kprobe(descCS)
+		descCS := C.CString(k)
+		ret, err := C.bpf_detach_kprobe(descCS)
+		if ret != 0 {
+			fmt.Errorf("fail to detach kprobe, %s", err)
```

Comment:
This would have no effect, as [`fmt.Errorf`](https://golang.org/pkg/fmt/#Errorf) is used to compose an `error`, not to print something. As we don't return any error from `Close()` currently, I would remove the check entirely.

Considering the comment above, the fixed version would look like:

```
        for k, v := range bpf.kprobes {
                C.perf_reader_free(v)
                evNameCS := C.CString(k)
                C.bpf_detach_kprobe(evNameCS)
                C.free(unsafe.Pointer(evNameCS))
        }
```